streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -542,6 +552,39 @@ public void shouldPropagateExceptionFromStandbyCommit() {
         verify(standby);
     }
 
+    @Test
+    public void shouldSendPurgeData() {
+        final KafkaFuture<DeletedRecords> futureDeletedRecords = new KafkaFutureImpl<>();
+        final Map<TopicPartition, RecordsToDelete> recordsToDelete = Collections.singletonMap(t1p1, RecordsToDelete.beforeOffset(5L));
+        final DeleteRecordsResult deleteRecordsResult = new DeleteRecordsResult(Collections.singletonMap(t1p1, futureDeletedRecords));
+
+        EasyMock.expect(active.recordsToDelete()).andReturn(recordsToDelete).once();
+        EasyMock.expect(adminClient.deleteRecords(recordsToDelete)).andReturn(deleteRecordsResult).once();
+        replay();
+
+        taskManager.maybePurgeCommitedRecords();
+        // second call should be no-op as the previous one is not done yet
+        taskManager.maybePurgeCommitedRecords();
+        verify(active, adminClient);
+    }
+
+    @Test
+    public void shouldIgnorePurgeDataErrors() {
+        final KafkaFutureImpl<DeletedRecords> futureDeletedRecords = new KafkaFutureImpl<>();
+        final Map<TopicPartition, RecordsToDelete> recordsToDelete = Collections.singletonMap(t1p1, RecordsToDelete.beforeOffset(5L));
+        final DeleteRecordsResult deleteRecordsResult = new DeleteRecordsResult(Collections.singletonMap(t1p1, (KafkaFuture<DeletedRecords>) futureDeletedRecords));
```

Comment:
Cast unnecessary?

Reply:
Unfortunately it is not, since we are using `KafkaFutureImpl` for this test (but not the other) in order to call `completeExceptionally`.

Reply:
Can't we specify the type via `Collections.<KafkaFuture>singletonMap(..)` to make it work? A Map of type `KafkaFuture` should accept `KafkaFutureImpl` -- or do I miss something? Generics can be confusion sometimes :)